artifact/parser.cpp

```diff
@@ -121,6 +121,13 @@ ExpectedArtifact Parse(io::Reader &reader, config::ParserConfig config) {
 	tok = lexer.Next();
 	optional::optional<ManifestSignature> signature;
 	if (tok.type == token::Type::ManifestSignature) {
+		// Early check for keys
+		if (config.artifact_verify_keys.size() == 0) {
+			return expected::unexpected(parser_error::MakeError(
+				parser_error::Code::ParseError,
+				"Missing public key(s) to verify manifest signature"));
```

Comment:
Missing keys is not an error. It's allowed to install a signed artifact onto a device which doesn't have signatures enabled, but not vice versa.

Reply:
Good catch. Thanks